components/tidb_query_expr/src/impl_time.rs

```diff
@@ -865,6 +865,12 @@ pub fn duration_string_time_diff(
     duration_duration_time_diff(ctx, arg1, &arg2)
 }
 
+#[rpn_fn(nullable)]
```

Comment:
I think it's ok to just use `#[rpn_fn]` and let this function accepts an `&DateTime`.